pkg/cmd/cli/cmd/login.go

```diff
@@ -0,0 +1,250 @@
+							}
+						}
+					}
+
+				}
+				if size > 1 {
+					projectsNames := make([]string, size)
+					for i, project := range projects.Items {
+						projectsNames[i] = project.Name
+					}
+					fmt.Printf("Your projects are: %v. You can switch between them at any time using 'osc project <project-name>'.\n", strings.Join(projectsNames, ", ")) // TODO switch projects TODO parameterize cmd name
+				}
+				current, err := f.OpenShiftClientConfig.Namespace()
+				checkErr(err)
+				fmt.Printf("Now using project '%v'\n", current)
+			} else {
+				// TODO handle no projects
+			}
+		},
+	}
+
+	cmds.Flags().StringVarP(&options.username, "username", "u", "", "Username, will prompt if not provided")
+	cmds.Flags().StringVarP(&options.password, "password", "p", "", "Password, will prompt if not provided")
+	cmds.Flags().StringVar(&options.project, "project", "", "If provided the client will switch to use the provided project after logging in")
+	// TODO should *probably* explicitly expose the flags below as local (currently global)
+	//cmds.Flags().StringVar(&options.context, "context", "", "Use a specific config context to log in")
+	//cmds.Flags().StringVar(&options.server, "server", "", "The address and port of the API server to log in to")
+	return cmds
+}
+
+func whoami(clientCfg *kclient.Config) (string, error) {
+	oClient, err := client.New(clientCfg)
+	if err != nil {
+		return "", err
+	}
+
+	me, err := oClient.Users().Get("~")
+	if err != nil {
+		return "", err
+	}
+
+	return me.FullName, nil
+}
+
+func (o *loginOptions) validate(configStore *config.ConfigStore, clientCfg *kclient.Config) error {
+	if o.contextProvided() {
+		ctx, ctxExists := configStore.Config.Contexts[o.context]
+
+		// context must exist
+		if !ctxExists {
+			return fmt.Errorf("Context '%s' not found\n", o.context)
+		}
+
+		// if context provides a given information then that flag is mutually exclusive
+		if (o.usernameProvided() || o.passwordProvided()) && len(ctx.AuthInfo) > 0 {
+			return fmt.Errorf("The 'context' flag cannot be used with 'username' or 'password' since it already provides these information. You must either provide a context or username and password.")
+		}
+		if o.serverProvided() && len(ctx.Cluster) > 0 {
+			return fmt.Errorf("The 'context' flag cannot be used with 'server' since it already provides this information. You must either provide a context or server.")
+		}
+		if o.projectProvided() && len(ctx.Namespace) > 0 {
+			return fmt.Errorf("The 'context' flag cannot be used with 'project' since it already provides this information. You must either provide a context or project.")
+		}
+
+		// if context were provided, take everything from it
+		o.username = ctx.AuthInfo
+		o.password = ""
+		o.server = ctx.Cluster
+		o.project = ctx.Namespace
+	}
+
+	// project must exist if provided
+	if o.projectProvided() {
+		oClient, err := client.New(clientCfg)
+		if err != nil {
+			return err
+		}
+		_, err = oClient.Projects().Get(o.project)
```

Comment:
If I'm unauthorized this call will produce an error, but you're calling `validate` before the user has had an opportunity to provide credentials.